src/components/Message/index.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react'
+import classNames from 'classnames'
+
+import Button from '../Button'
+import ButtonGroup from '../ButtonGroup'
+import Input from '../Form/Input'
+import styles from './styles.less'
+
+export default class Message extends Component {
+   static propTypes = {
+      onConfirm: React.PropTypes.func,
+      onPrompt: React.PropTypes.func,
+      text: React.PropTypes.string,
+      title: React.PropTypes.string,
+      type: React.PropTypes.string,
+      showing: React.PropTypes.bool
+   }
+
+   static defaultProps = {
+      text: 'Text',
+      title: 'Title',
+      type: 'alert',
+      showing: false
+   }
+
+   constructor(props) {
+      super(props)
+
+      this.close = this.close.bind(this)
+      this.handleCancelClick = this.handleCancelClick.bind(this)
+      this.handleOKClick = this.handleOKClick.bind(this)
+      this.handlePromptInputChange = this.handlePromptInputChange.bind(this)
+
+      this.state = {
+         promptInputValue: '',
+         showing: props.showing
+      }
+   }
+
+   componentWillReceiveProps(nextProps) {
+      this.setState({
+         showing: nextProps.showing
+      })
+   }
+
+   close() {
+      this.setState({ showing: false })
+   }
+
+   handleCancelClick() {
+      if (this.props.type === 'confirm') {
+         this.props.onConfirm(false)
+      } else if (this.props.type === 'prompt') {
+         this.setState({ promptInputValue: '' })
+      }
+      this.close()
+   }
+
+   handleOKClick() {
+      if (this.props.type === 'confirm') {
+         this.props.onConfirm(true)
+      } else if (this.props.type === 'prompt') {
+         this.props.onPrompt(this.state.promptInputValue)
+         this.setState({ promptInputValue: '' })
+      }
+      this.close()
+   }
+
+   handlePromptInputChange(event) {
+      this.setState({
+         promptInputValue: event.target.value
+      })
+   }
+
+   render() {
+      const displayClass = this.state.showing ? 'showMessage' : 'hideMessage'
+      const display = this.state.showing ? 'table' : 'none'
+
```

Comment:
As for showing -> visible, I will make a separate PR that changes this on both this component and the Notification component as well, since `showing` is used there too. And communities will have to be updated at the same time to use `visible`.